obspy/io/sac/sactrace.py

```diff
@@ -1092,7 +1175,7 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                 val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ascii', 'replace')
```

Comment:
Hopefully, [this comment](https://github.com/obspy/obspy/pull/1773#issuecomment-299992461) should help you update your branch.